examples/acoustic/Acoustic_codegen.py

```diff
@@ -95,40 +63,37 @@ def Forward(self, save=False, cache_blocking=None, auto_tuning=False,
                 dse='advanced', dle='advanced', compiler=None, u_ini=None, legacy=True):
         nt, nrec = self.data.shape
         nsrc = self.source.shape[1]
-        ndim = len(self.damp.shape)
+        ndim = len(self.model.shape)
         h = self.model.get_spacing()
-        dtype = self.damp.dtype
+        dtype = self.model.dtype
         nbpml = self.model.nbpml
-        dt = self.model.get_critical_dt()
-        if self.t_order == 4:
-            dt *= 1.73
 
         # Create source symbol
         p_src = Dimension('p_src', size=nsrc)
         src = SourceLike(name="src", dimensions=[time, p_src], npoint=nsrc, nt=nt,
-                         dt=dt, h=h, ndim=ndim, nbpml=nbpml, dtype=dtype,
+                         dt=self.dt, h=h, ndim=ndim, nbpml=nbpml, dtype=dtype,
                          coordinates=self.source.receiver_coords)
         src.data[:] = self.source.traces[:]
 
         # Create receiver symbol
         p_rec = Dimension('p_rec', size=nrec)
         rec = SourceLike(name="rec", dimensions=[time, p_rec], npoint=nrec, nt=nt,
-                         dt=dt, h=h, ndim=ndim, nbpml=nbpml, dtype=dtype,
+                         dt=self.dt, h=h, ndim=ndim, nbpml=nbpml, dtype=dtype,
                          coordinates=self.data.receiver_coords)
 
```

Comment:
O hadn't realized, but when did the "remove 2nd dim" for 2D disapeared. In practice, even for 2D it is quite usual to have still 3D coordinates wit h0 everywhere in z which would break in our case as dimensions won't match.